### ISSUE TITLE
Fix warnings

### ISSUE THREAD

**Issue by jlnr** [gosu/gosu#355]
Something that can be done after #354: Fix the gazillion warnings that come up when compiling Gosu.

Example attachment from #352: https://gist.github.com/anonymous/e37567116ccedfe8ffeb08a938dfd916

Many of these warnings are also visible in the (expanded) logs from Travis and Appveyor:

https://travis-ci.org/gosu/gosu/jobs/187743955 (macOS)
https://travis-ci.org/gosu/gosu/jobs/187753484 (Linux)
https://ci.appveyor.com/project/gosu-ci/gosu/build/159/job/9c5r5sr8284pda67 (Windows, not too bad)

**Comment by jlnr** [comment]
Duplicate of #197, will close this for the same reason.
